### DISCUSSION TITLE
Upgrade fails when run in a container

### DISCUSSION THREAD

**Discussion by rmenti74** [globaleaks/globaleaks-whistleblowing-software#3860] [Q&A]
I tried to update to the latest version from within a Docker container. When using the command 'apt-get install globaleaks,' I encountered this error:

root@1ecbfb1db03a:/# apt-get install globaleaks
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages will be upgraded:
  globaleaks
1 upgraded, 0 newly installed, 0 to remove and 13 not upgraded.
1 not fully installed or removed.
Need to get 82.6 MB of archives.
After this operation, 6144 B of additional disk space will be used.
Get:1 https://deb.globaleaks.org bookworm/ globaleaks 4.13.20 [82.6 MB]
Fetched 82.6 MB in 1s (72.0 MB/s)     
debconf: delaying package configuration, since apt-utils is not installed
(Reading database ... 18482 files and directories currently installed.)
Preparing to unpack .../globaleaks_4.13.20_all.deb ...
swapoff: Not superuser.
Stopping ...Disabling GlobaLeaks Network Sandboxing...iptables-save v1.8.9 (nf_tables): Could not fetch rule set generation id: Permission denied (you must be root)
ip6tables-save v1.8.9 (nf_tables): Could not fetch rule set generation id: Permission denied (you must be root)
done.

When using the command './install-globaleaks.sh,' I received:

root@1ecbfb1db03a:/# ./install-globaleaks.sh
Running the GlobaLeaks installation...
In case of failure please report encountered issues to the ticketing system at: https://github.com/globaleaks/GlobaLeaks/issues

Detected OS: Debian - bookworm
Running: "apt-get -y update"... SUCCESS
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
tzdata is already the newest version (2023c-5+deb12u1).
0 upgraded, 0 newly installed, 0 to remove and 13 not upgraded.
1 not fully installed or removed.
Need to get 82.6 MB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 https://deb.globaleaks.org bookworm/ globaleaks 4.13.18 [82.6 MB]
Fetched 82.6 MB in 1s (66.6 MB/s)     

Current default time zone: 'Etc/UTC'
Local time is now:      Mon Dec 11 10:53:15 UTC 2023.
Universal Time is now:  Mon Dec 11 10:53:15 UTC 2023.

Running: "apt-get -y install gnupg net-tools software-properties-common wget"... SUCCESS
Running: "is_tcp_sock_free_check 0.0.0.0:80"... SUCCESS
Running: "is_tcp_sock_free_check 0.0.0.0:443"... SUCCESS
Running: "is_tcp_sock_free_check 127.0.0.1:8082"... SUCCESS
Running: "is_tcp_sock_free_check 127.0.0.1:8083"... FAIL
Ouch! The installation failed.
COMBINED STDOUT/STDERR OUTPUT OF FAILED COMMAND:

How can it be resolved?
Thk

**Comment by evilaliv3** [comment]
Hello,

would you please try with the up to date install script?

**Reply by rmenti74** [reply]
with the up to date script executed within container (docker exec -u root -it <container-id> /bin/bash) i get:

root@1ecbfb1db03a:/# ./install-globaleaks.sh
Running the GlobaLeaks installation...
In case of failure please report encountered issues to the ticketing system at: https://github.com/globaleaks/GlobaLeaks/issues

Detected OS: Debian - bookworm
Running: "apt-get -y update"... SUCCESS
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
tzdata is already the newest version (2023c-5+deb12u1).
0 upgraded, 0 newly installed, 0 to remove and 13 not upgraded.
1 not fully installed or removed.
Need to get 82.6 MB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 https://deb.globaleaks.org bookworm/ globaleaks 4.13.18 [82.6 MB]
Fetched 82.6 MB in 1s (76.4 MB/s)     

Current default time zone: 'Etc/UTC'
Local time is now:      Mon Dec 11 13:08:15 UTC 2023.
Universal Time is now:  Mon Dec 11 13:08:15 UTC 2023.

Running: "apt-get -y install gnupg net-tools software-properties-common wget"... SUCCESS
Adding GlobaLeaks PGP key to trusted APT keys
Updating GlobaLeaks apt source.list in /etc/apt/sources.list.d/globaleaks.list ...
Running: "apt-get update -y"... SUCCESS
Running: "apt-get install globaleaks -y"... root@node34919-env-3327245 ~ $ 

and exit from container with no error (root@node34919-env-3327245 ~ $)

if I repeat the same operations:

root@1ecbfb1db03a:/# ./install-globaleaks.sh
Running the GlobaLeaks installation...
In case of failure please report encountered issues to the ticketing system at: https://github.com/globaleaks/GlobaLeaks/issues

Detected OS: Debian - bookworm
Running: "apt-get -y update"... SUCCESS
E: dpkg was interrupted, you must manually run 'dpkg --configure -a' to correct the problem. 

Current default time zone: 'Etc/UTC'
Local time is now:      Mon Dec 11 13:17:46 UTC 2023.
Universal Time is now:  Mon Dec 11 13:17:46 UTC 2023.

Running: "apt-get -y install gnupg net-tools software-properties-common wget"... FAIL
Ouch! The installation failed.
COMBINED STDOUT/STDERR OUTPUT OF FAILED COMMAND:
E: dpkg was interrupted, you must manually run 'dpkg --configure -a' to correct the problem. 
root@1ecbfb1db03a:/#

**Reply by rmenti74** [reply]
hello,
i have tried this command within docker container:

service globaleaks stop

with this error result:

Stopping ...Disabling GlobaLeaks Network Sandboxing...iptables-save v1.8.9 (nf_tables): Could not fetch rule set generation id: Permission denied (you must be root)
ip6tables-save v1.8.9 (nf_tables): Could not fetch rule set generation id: Permission denied (you must be root)
done.

hope this help.
thk

**Comment by Neovoncarstein** [comment]
I have the same problema, exactly the same... and no solution found. I can not update,
